src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,11 +1,30 @@
-export const TodoFilter = () => (
+import React, { Dispatch, SetStateAction } from 'react';
+import { Filter } from '../../types/Filter';
+
+type TodoFilterProps = {
+  query: string;
+  setQuery: Dispatch<SetStateAction<string>>;
+  filter: Filter;
+  setFilter: Dispatch<SetStateAction<Filter>>;
+};
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  query,
+  setQuery,
+  filter,
+  setFilter,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
+        <select
+          data-cy="statusSelect"
+          value={filter}
+          onChange={event => setFilter(event.target.value as Filter)}
```

Comment:
The `onChange` event handler uses type casting (`as Filter`) which can be unsafe if the value is not a valid `Filter` type. Consider using a type guard or a function to ensure the value is a valid `Filter` before setting it.